coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java

```diff
@@ -80,9 +83,14 @@ public ModelAndView start(@RequestParam("budgetId") Long budgetId, @RequestParam
         ModelAndView result = getModelAndViewService().getModelAndViewWithInit(form, ProposalBudgetConstants.KradConstants.BUDGET_DEFAULT_VIEW);
         form.setViewOnly(inViewMode);
         form.setCanEditView(!inViewMode);
-        if (budgetRatesService.checkActivityTypeChange(form.getBudget().getBudgetRates(), form.getDevelopmentProposal().getActivityTypeCode())) {
+    	boolean canModify = false;
+    	if(!inViewMode) {
+    		canModify = getProposalBudgetAuthorizer().canModifyBudget(form.getBudget(), getGlobalVariableService().getUserSession().getPerson());
```

Comment:
So KIM checks are already run, probably multiple times and can be expensive. You should be able to just check for this in the editModes on the form?